pyo3-macros-backend/src/pyfunction/signature.rs

```diff
@@ -624,8 +647,8 @@ impl<'a> FunctionSignature<'a> {
             output.push_str(parameter);
 
             if i >= py_sig.required_positional_parameters {
-                // has a default, just use ... for now
-                output.push_str("=...");
+                output.push('=');
+                output.push_str(&self.default_value_for_parameter(parameter));
```

Comment:
So because `self.default_value_for_parameter` does a search of the arguments by name, this is O(n²), but only for small n, so it's probably fine.

I was planning to avoid that by storing the default expressions in `self.python_signature`, but that was harder with the deprecated args attribute still present.

Reply:
> So because `self.default_value_for_parameter` does a search of the arguments by name, this is O(n²), but only for small n, so it's probably fine.

Yeah, I think the same.